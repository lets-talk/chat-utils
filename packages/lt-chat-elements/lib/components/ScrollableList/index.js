import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { withAutoScroll } from '../../utils/hoc';

const StyledScrollableListContainer = styled.div`
  max-height: 100vh;
  background-color: ${(props) => props.theme.palette.common.white};
  overflow-y: auto;
  padding: 10px;
`;

const ScrollableList = (props) => {
  const {
    cmpRef, className,
  } = props;
  return (
    <StyledScrollableListContainer
      innerRef={cmpRef}
      className={className}
      id="messages"
    >
      { props.children }
    </StyledScrollableListContainer>
  );
};

ScrollableList.displayName = 'ScrollableList';

ScrollableList.propTypes = {
  children: PropTypes.array,
  cmpRef: PropTypes.func,
  className: PropTypes.string,
};

ScrollableList.defaultProps = {
  cmpRef: null,
  className: 'LT-ScrollableList-Container',
};

const autoScrollOptions = { threshold: 300, direction: 'bottom' };
export default withAutoScroll(autoScrollOptions)(ScrollableList);
