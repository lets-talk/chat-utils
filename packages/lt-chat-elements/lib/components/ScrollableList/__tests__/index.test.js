import React from 'react';
import { shallow } from 'enzyme';

import ScrollableList from '../index';

describe('<ScrollableList /> component', () => {
  it('matches snapshot', () => {
    const list = shallow(<ScrollableList><div>Child 1</div><div>Child 2</div></ScrollableList>);
    expect(list).toMatchSnapshot();
  });

  it('renders children inside', () => {
    const list = shallow(<ScrollableList><div>Child 1</div><div>Child 2</div></ScrollableList>);
    expect(list.find('div').length).toBe(2);
  });
});
