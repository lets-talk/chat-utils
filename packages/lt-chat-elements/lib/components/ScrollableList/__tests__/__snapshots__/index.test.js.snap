// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollableList /> component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_class>
    <div>
      Child 1
    </div>
    <div>
      Child 2
    </div>
  </_class>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          Child 1
        </div>,
        <div>
          Child 2
        </div>,
      ],
      "className": "LT-ScrollableList-Container",
      "cmpRef": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Child 1",
        },
        "ref": null,
        "rendered": "Child 1",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Child 2",
        },
        "ref": null,
        "rendered": "Child 2",
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            Child 1
          </div>,
          <div>
            Child 2
          </div>,
        ],
        "className": "LT-ScrollableList-Container",
        "cmpRef": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Child 1",
          },
          "ref": null,
          "rendered": "Child 1",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Child 2",
          },
          "ref": null,
          "rendered": "Child 2",
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
